git-cms-checkdeps

```diff
@@ -1,288 +1,293 @@
+    if path.isfile(depfile):
+        readDependencyInfo(depfile, vals)
 
-sub readDependencyInfo()
-{
-  my $file=shift;
-  my $cache=shift;
-  my $ref;
-  open($ref,"gunzip -c $file |") || die "Can not open file for reading: $file\n";
-  while  (my $line=<$ref>) { 
-    chomp($line);
-    my ($file1,$rest)=split(' ',$line,2);
-    if ($file1 ne "")
-    {
-      if (!exists $vals{$file1}){$vals{$file1}=[];}
-      push @{$vals{$file1}},split(' ',$rest);
-    }
-  }
-  close($ref);
-}
+chdir("%s/src" % localtop)
+reltag = None
+if environ.get("CMSSW_GIT_HASH"):
+    reltag = environ["CMSSW_GIT_HASH"]
+else:
+    reltag = environ["CMSSW_VERSION"]
 
-sub poisonIncludes() {
-    my ($deletedFiles,$topdir)=@_;
-    use File::Path qw (mkpath rmtree);
-    use File::Basename qw(dirname);
-    my $poison="${topdir}/${poisondir}";
-    if (-d $poison){rmtree($poison);}
-    my $poisondata = "$poison/.data";
-    mkpath($poisondata);
-    if (scalar(@$deletedFiles)>0) {print ">> Creating dummy files under $poison directory.\n";}
-    foreach my $file (@$deletedFiles) {
-        my $f="${poison}/${file}";
-        my $dir=dirname($f);
-        if (!-d $dir){mkpath($dir);}
-        if ($file=~/\/data\//o){symlink("$poisondata","$f");}
-        else{
-          my $ref;
-          if(!open($ref,">${f}")){die "ERROR: Can not open file for writing: $f\n";}
-          print $ref "#error THIS FILE HAS BEEN REMOVED FROM THE PACKAGE.\n";
-          close($ref);
-        }
-        print "   $file\n";
-    }
-}
+args_0 = ['git', 'diff', '''-G^([^$]+$|[^$]*[$][^$]*$|([^$]*[$])($|[^RAIDLNSH]|(R[^eC]|A[^u]|I[^d]|D[^a]|L[^o]|
+    N[^a]|S[^o]|H[^e])|(Re[^v]|RC[^S]|Au[^t]|Id[^:]|Da[^t]|Lo[^g]|Na[^m]|So[^u]|He[^a])|(Rev[^i]|RCS[^f]|Aut[^h]|
+    Dat[^e]|Log[^:]|Nam[^e]|Sou[^r]|Hea[^d])|(Revi[^s]|RCSf[^i]|Auth[^o]|Date[^:]|Name[^:]|Sour[^c]|Head[^e])|
+    (Revis[^i]|RCSfi[^l]|Autho[^r]|Sourc[^e]|Heade[^r])|(Revisi[^o]|RCSfil[^e]|Author[^:]|Source[^:]|Header[^:])|
+    (Revisio[^n]|RCSfile[^:])|(Revision[^:])))''', '--name-status', '-r', reltag]
+args_1 = ['grep', '-v', '.gitignore']
+args_2 = ['sed', '-e', 's/[ \\t]\\+/ /']
+args_3 = ['git', 'diff', '--diff-filter', 'R', '--name-status', '-r', reltag]
+pipe_0 = subprocess.Popen(args_0, stdout=subprocess.PIPE)
+pipe_1 = subprocess.Popen(args_1, stdin=pipe_0.stdout, stdout=subprocess.PIPE)
+pipe_2 = subprocess.Popen(args_2, stdin=pipe_1.stdout, stdout=subprocess.PIPE)
+pipe_3 = subprocess.Popen(args_3, stdout=subprocess.PIPE)
+if not pipe_3.communicate()[0]:
+    stdout = pipe_2.communicate()[0].splitlines()
+else:
+    stdout = pipe_3.communicate()[0].splitlines()
+for diff in stdout:
+    diff = diff.decode()
+    match = re.search(
+        r'^[MUDR]([0-9]*)\s+([^\/]+\/[^\/]+)\/[^\s]+(\s+[^\/]+\/[^\/]+\/[^\s]+|)$', diff)
+    if match:
+        pack = match.group(2)
+        if pack not in upackages:
+            upackages[pack] = []
+            if not pack:
```

Comment:
this is wrong. In perl code we are checking for the existance of the package on filesystem